### PR TITLE
Add debug fn for program ids

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Nice to see when there is an error what program it is from.

#### Summary of Changes

Add function to map ids to a string name, print that when there is an error.

Fixes #

### CODE REVIEW

File: src/bank.rs [hunk 1]
```diff
@@ -412,6 +412,19 @@ impl Bank {
         Ok(())
     }
 
+    fn program_id_to_string(id: &Pubkey) -> String {
+        let mut map = HashMap::new();
```
Comment (mvines) [inline src/bank.rs]: It would be nice to not receate this hashmap on every call

Comment (sakridge) [reply src/bank.rs]: Yea, I agree. Seems the options are use `static mut` with an unsafe block, potentially a crate called phf which is like a perfect hashing library or use an if-else chain. I tried with match but I didn't get it to work. None of the options seemed that good.

Comment (sakridge) [reply src/bank.rs]: hmm, phf is nightly-only. lazy_static is another option which I *think* should work on stable.

**Comment by sakridge** [comment]
@garious @mvines ping

**Comment by garious** [comment]
I'd prefer instead that every program advertise a `name()` function similar to how it already advertises an `entrypoint()` function using the entrypoint macro.  We should avoid anything around the notion that there's some fixed set of builtin programs. That's a wart in the current implementation. Nearly all of the programs should be boot from the bank.

**Comment by garious** [comment]
@sakridge, close or finish this one?

**Pull request closed without merging**